tests/pipelines/test_predict_step.py

```diff
@@ -101,7 +103,68 @@ def test_predict_step_runs(
     )
     predict_step._run(str(predict_step_output_dir))
 
-    prediction_assertions(predict_step_output_dir, "parquet", "scored", spark_session)
+    # Test internal predict step output artifact
+    artifact_file_name, artifact_file_extension = _SCORED_OUTPUT_FILE_NAME.split(".")
+    prediction_assertions(
+        predict_step_output_dir, artifact_file_extension, artifact_file_name, spark_session
+    )
+    # Test user specified output
+    prediction_assertions(predict_step_output_dir, "parquet", "output", spark_session)
+
+
+def test_predict_step_uses_register_step_model_name(
+    tmp_pipeline_root_path: Path,
+    predict_step_output_dir: Path,
+    spark_session,
+):
+    rm_name = "register_step_model"
+    train_log_and_register_model(rm_name, is_dummy=True)
+
+    predict_step = PredictStep.from_pipeline_config(
+        {
+            "steps": {
+                "register": {"model_name": rm_name},
+                "predict": {
+                    "output_format": "parquet",
+                    "output_location": str(predict_step_output_dir.joinpath("output.parquet")),
+                    "_disable_env_restoration": True,
+                },
+            }
+        },
+        str(tmp_pipeline_root_path),
+    )
+    predict_step._run(str(predict_step_output_dir))
+
+    prediction_assertions(predict_step_output_dir, "parquet", "output", spark_session)
+
+
+def test_predict_model_uri_takes_precendence_over_model_name(
```

Comment:
Where is the assertion verifying `model_uri` is actually being used?

Reply:
Yeah, this could be a bit confusing to read. This test trains and registers two models. One normal regression model and one dummy model. It gives the register step the model name of the regression model and the predict step the model uri of the dummy model. The assertions on the scored dataset only pass if the dummy model was used in the scoring.

Should I add some comments to clarify this?

Reply:
I'll add some clarifying comments anyways and merge this PR.